metadata-ingestion/src/datahub_provider/_plugin.py

```diff
@@ -290,12 +289,35 @@ def custom_on_success_callback(context):
     return custom_on_success_callback
 
 
-def task_policy(task: BaseOperator) -> None:
+def task_policy(task: Union[BaseOperator, MappedOperator]) -> None:
     task.log.debug(f"Setting task policy for Dag: {task.dag_id} Task: {task.task_id}")
     # task.add_inlets(["auto"])
     # task.pre_execute = _wrap_pre_execution(task.pre_execute)
-    task.on_failure_callback = _wrap_on_failure_callback(task.on_failure_callback)
-    task.on_success_callback = _wrap_on_success_callback(task.on_success_callback)
+
+    # MappedOperator's callbacks don't have setters until Airflow 2.X.X
+    # https://github.com/apache/airflow/issues/24547
+    # We can bypass this by going through partial_kwargs for now
+    if MappedOperator and isinstance(task, MappedOperator):  # type: ignore
+        on_failure_callback_prop: property = getattr(
+            MappedOperator, "on_failure_callback"
+        )
+        on_success_callback_prop: property = getattr(
+            MappedOperator, "on_success_callback"
+        )
+        if not on_failure_callback_prop.fset or not on_success_callback_prop.fset:
+            task.log.debug(
+                "Using MappedOperator's partial_kwargs instead of callback properties"
+            )
+            task.partial_kwargs["on_failure_callback"] = _wrap_on_failure_callback(
+                task.on_failure_callback
+            )
+            task.partial_kwargs["on_success_callback"] = _wrap_on_success_callback(
```

Comment:
Minor: Looks like we reuse the raw strings a couple times:

- on_success_callback
- on_failure_callback

can we extract into a const?

Reply:
Done